cli/command/task.go

```diff
@@ -10,6 +17,219 @@ func NewTaskCommand(b *CommandBase) *TaskCommand {
+					},
+					cli.StringFlag{
+						Name:  "start",
+						Usage: "the start of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+					cli.StringFlag{
+						Name:  "end",
+						Usage: "the end of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (t *TaskCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "ENVIRONMENT", "TASK_NAME", "DEPLOY")
+	if err != nil {
+		return err
+	}
+
+	overrides, err := parseOverrides(c.StringSlice("env"))
+	if err != nil {
+		return err
+	}
+
+	environmentID, err := t.resolveSingleEntityIDHelper("environment", args["ENVIRONMENT"])
+	if err != nil {
+		return err
+	}
+
+	deployID, err := t.resolveSingleEntityIDHelper("deploy", args["DEPLOY"])
+	if err != nil {
+		return err
+	}
+
+	req := models.CreateTaskRequest{
+		TaskName:           args["TASK_NAME"],
+		EnvironmentID:      environmentID,
+		DeployID:           deployID,
+		ContainerOverrides: overrides,
+	}
+
+	jobID, err := t.client.CreateTask(req)
+	if err != nil {
+		return err
+	}
+
+	return t.waitOnJobHelper(c, jobID, "creating", func(taskID string) error {
+		task, err := t.client.ReadTask(taskID)
+		if err != nil {
+			return err
+		}
+
+		return t.printer.PrintTasks(task)
+	})
+}
+
+func (t *TaskCommand) delete(c *cli.Context) error {
+	return t.deleteHelper(c, "task", func(taskID string) (string, error) {
+		return t.client.DeleteTask(taskID)
+	})
+}
+
+func (t *TaskCommand) list(c *cli.Context) error {
+	taskSummaries, err := t.client.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	return t.printer.PrintTaskSummaries(taskSummaries...)
+}
+
+func (t *TaskCommand) read(c *cli.Context) error {
+	taskSummaries, err := t.client.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	taskExists := map[string]bool{}
+	for _, taskSummary := range taskSummaries {
+		taskExists[taskSummary.TaskID] = true
+	}
+
+	args, err := extractArgs(c.Args(), "TASK_NAME")
+	if err != nil {
+		return err
+	}
+
+	taskIDs, err := t.resolver.Resolve("task", args["TASK_NAME"])
+	if err != nil {
+		return err
+	}
+
+	tasks := make([]*models.Task, len(taskIDs))
+	for i, taskID := range taskIDs {
+		if !taskExists[taskID] {
+			log.Printf("[DEBUG] Resolver returned an expired task '%s'", taskID)
+			continue
+		}
+
+		if taskExists[taskID] {
+			task, err := t.client.ReadTask(taskID)
+			if err != nil {
+				return err
+			}
+
+			tasks[i] = task
+		}
+	}
+
+	return t.printer.PrintTasks(tasks...)
+}
+
+func (t *TaskCommand) logs(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "TASK_NAME")
+	if err != nil {
+		return err
+	}
+
+	taskID, err := t.resolveSingleEntityIDHelper("task", args["TASK_NAME"])
+	if err != nil {
+		return err
+	}
+
+	query := buildLogQueryHelper(taskID, c.String("start"), c.String("end"), c.Int("tail"))
+	logs, err := t.client.ReadTaskLogs(taskID, query)
+	if err != nil {
+		return err
+	}
+
+	return t.printer.PrintLogs(logs...)
```

Comment:
[nit] This can take a while to return as well, consider adding the job and/or spinner behavior